server.go

```diff
@@ -138,23 +142,25 @@ func NewFromConfig(conf Config) (ret Server, err error) {
 		})
 
 		if err != nil {
-			ret.logger.Info("error getting AWS session: %s", err)
+			log.Info("error getting AWS session: %s", err)
 			svc = nil
 		} else {
-			ret.logger.Info("Successfully created AWS session")
+			log.Info("Successfully created AWS session")
 			svc = s3.New(sess)
 
 			// TODO(aditya) store this on the server
 			S3Bucket = conf.AWSBucket
 		}
 	} else {
-		ret.logger.Info("AWS credentials not found")
+		log.Info("AWS credentials not found")
 	}
 
 	if svc == nil {
-		ret.logger.Info("S3 archives are disabled")
+		plugin := S3Plugin{svc: svc}
+		plugin.Initialize(ret.statsd, log)
```

Comment:
what's the argument for having a separate Initialize method that just sets the two fields? is that method also going to become a standard part of the plugin interface at some point? otherwise it seems simpler to me to just declare the whole object in one expression